providers/implementations/kdfs/kbkdf.c

```diff
@@ -186,8 +188,21 @@ static int derive(EVP_MAC_CTX *ctx_init, kbkdf_mode mode, unsigned char *iv,
         if (mode == FEEDBACK && !EVP_MAC_update(ctx, k_i, k_i_len))
             goto done;
 
-        if (!EVP_MAC_update(ctx, (unsigned char *)&i, 4)
-            || !EVP_MAC_update(ctx, label, label_len)
+        if (r == 32) {
```

Comment:
Is this more readable:
```c
if (!EVP_MAC_update(ctx, 4 - r / 8 + (unsigned char *)&i, r / 8))
    goto done;
```
It's more concise and avoids the if-spaghetti.

Reply:
Thanks for that suggestion!